### ISSUE TITLE
`Regexp` check for unnecessary `// ok` comments is configured under `TreeWalker`

### ISSUE THREAD

**Issue by smita1078** [checkstyle/checkstyle#19993]
## Description

While working on https://github.com/checkstyle/checkstyle/pull/19983 , we noticed that plain `// ok` comments are currently not reporting violations even though there is an `UnnecessaryOkComment` regexp check configured.

Current configuration:

```xml id="5ekn6o"
<module name="TreeWalker">
    ...
    <module name="Regexp">
      <property name="id" value="UnnecessaryOkComment"/>
      <property name="format"
               value="(?i)\/\/\s*ok$"/>
      <property name="illegalPattern" value="true"/>
      <message key="illegal.regexp"
               value="'// ok' comments should have an explanation, or they are unnecessary."/>
    </module>
</module>
```

However, `Regexp` is a Checker-level module and not a TreeWalker module.

Because of this, the check does not appear to be applied, which is why existing plain `// ok` comments in input files (for example some `ParenPad` inputs) are currently not producing violations.

## Expected Behavior

Plain comments like:

```java id="q6e8kg"
// ok
```

should report violations according to the configured regexp.

## Proposed Fix

Move the `Regexp` module from under `TreeWalker` to the `Checker` level configuration so it is properly applied repository-wide.

**Comment by romani** [comment]
It will be good to look why configuration loading did not throw exception on such situation. It is better to fail execution completely if any misconfigurations present.

**Comment by smita1078** [comment]
[Web-site automation 2025](https://github.com/orgs/checkstyle/projects/13) I think Web-site automation should be under Summer 2026 instead of 2025.

**Comment by smita1078** [comment]
> It will be good to look why configuration loading did not throw exception on such situation. It is better to fail execution completely if any misconfigurations present.

What I discovered on further debugging:

1. When moved Regexp as a child to Checker
```
[ERROR] C:\Users\Smita Prajapati\checkstyle\config\ant-phase-verify.xml:205: Unable to create Root Module: config {C:\Users\Smita Prajapati\checkstyle\config\checkstyle-input-checks.xml}.                                                                                                                                                                             
[ERROR] around Ant part ...<ant antfile="config/ant-phase-verify.xml" />... @ 7:50 in C:\Users\Smita Prajapati\checkstyle\target\antrun\build-main.xml: Regexp is not allowed as a child in Checker
```
Regexp is not allowed as a child in Checker
Regexp is not a Checker-level module either , it's actually a TreeWalker. That's why there was no such Exception thrown.

2. The `UnnecessaryOkComment `check was added to` checkstyle-input-checks.xml`, but the xdocs example files (src/xdocs-examples/resources/**/*.java) are never scanned by that config. The ant task that runs `checkstyle-input-checks.xml` explicitly excludes xdocs-examples/ from its `fileset`. Instead, xdocs example files have their own dedicated ant task that runs `checkstyle-examples-checks.xml`.
So the check existed but was running against the wrong set of files ,it never saw the files that contained the // ok comments.

On adding check to `checkstyle-examples-checks.xml`.

it worked fine with multiple errors coming into picture:
```
[INFO]      [echo] Checkstyle started (checkstyle-examples-checks.xml): 29/05/2026 10:01:26 pm
[INFO] [checkstyle] Running Checkstyle  on 793 files
[ERROR] [checkstyle] [ERROR] C:\Users\Smita Prajapati\checkstyle\src\xdocs-examples\resources\com\puppycrawl\tools\checkstyle\checks\annotation\annotationlocation\Example1.java:24: // ok comments should have an explanation, or they are unnecessary. [UnnecessaryOkComment]
[ERROR] [checkstyle] [ERROR] C:\Users\Smita Prajapati\checkstyle\src\xdocs-examples\resources\com\puppycrawl\tools\checkstyle\checks\annotation\annotationlocation\Example1.java:25: // ok comments should have an explanation, or they are unnecessary. [UnnecessaryOkComment]
[ERROR] [checkstyle] [ERROR] C:\Users\Smita Prajapati\checkstyle\src\xdocs-examples\resources\com\puppycrawl\tools\checkstyle\checks\annotation\annotationlocation\Example1.java:26: // ok comments should have an explanation, or they are unnecessary. [UnnecessaryOkComment]
[ERROR] [checkstyle] [ERROR] C:\Users\Smita Prajapati\checkstyle\src\xdocs-examples\resources\com\puppycrawl\tools\checkstyle\checks\annotation\annotationlocation\Example1.java:28: // ok comments 
....................
[INFO] ------------------------------------------------------------------------
```

**Comment by romani** [comment]
@smita1078 , thanks a lot for your analysis!!
https://checkstyle.org/checks/regexp/regexp.html#Parent_Module it is treewalker

**Comment by smita1078** [comment]
> [@smita1078](https://github.com/smita1078) , thanks a lot for your analysis!! https://checkstyle.org/checks/regexp/regexp.html#Parent_Module it is treewalker

Yes 👍
